plugins/modules/packaging/language/ansible_galaxy_install.py

```diff
@@ -0,0 +1,262 @@
+    description: The value of the I(name) parameter.
+    type: str
+    returned: always
+  dest:
+    description: The value of the I(dest) parameter.
+    type: str
+    returned: always
+  requirements_file:
+    description: The value of the I(requirements_file) parameter.
+    type: str
+    returned: always
+  force:
+    description: The value of the I(force) parameter.
+    type: bool
+    returned: always
+  roles:
+    description:
+    - If I(requirements_file) is specified instead, returns dictionary with all the roles installed per path.
+    - If I(name) is specified, returns that role name and the version installed per path.
+    type: dict
+    returned: always when installing roles
+    contains:
+      "<path>":
+        description: Roles and versions for that path
+        type: dict
+    sample:
+      /home/az/.ansible/roles:
+        ansistrano.deploy: 3.9.0
+        baztian.xfce: v0.0.3
+      /custom/ansible/roles:
+        ansistrano.deploy: 3.8.0
+  collections:
+    description:
+    - If I(requirements_file) is specified instead, returns dictionary with all the collections installed per path.
+    - If I(name) is specified, returns that collection name and the version installed per path.
+    type: dict
+    returned: always when installing collections
+    contains:
+      "<path>":
+        description: Collections and versions for that path
+        type: dict
+    sample:
+      /home/az/.ansible/collections/ansible_collections:
+        community.docker: 1.6.0
+        community.general: 3.0.2
+      /custom/ansible/ansible_collections:
+        community.general: 3.1.0
+"""
+
+import re
+import json
+
+from ansible_collections.community.general.plugins.module_utils.module_helper import CmdModuleHelper, ArgFormat
+
+
+class AnsibleGalaxyInstall(CmdModuleHelper):
+    output_params = ('type', 'name', 'dest', 'requirements_file', 'force')
+    module = dict(
+        argument_spec=dict(
+            type=dict(type='str', choices=('collection', 'role', 'both'), required=True),
+            name=dict(type='str'),
+            requirements_file=dict(type='path'),
+            dest=dict(type='path'),
+            force=dict(type='bool', default=False),
+        ),
+        mutually_exclusive=[('name', 'requirements_file')],
+        required_one_of=[('name', 'requirements_file')],
+        required_if=[('type', 'both', ['requirements_file'])],
+        supports_check_mode=False,
+    )
+
+    command = 'ansible-galaxy'
+    command_args_formats = dict(
+        type=dict(fmt=lambda v: [] if v == 'both' else [v]),
+        galaxy_cmd=dict(),
+        requirements_file=dict(fmt=('-r', '{0}'),),
+        dest=dict(fmt=('-p', '{0}'),),
+        force=dict(fmt="--force", style=ArgFormat.BOOLEAN),
+        format=dict(fmt=["--format", "json"], style=ArgFormat.BOOLEAN),
+    )
+    check_rc = True
+
+    def __list_collections__(self):
+        params = ({'type': 'collection'}, {'galaxy_cmd': 'list'}, 'dest', 'format')
```

Comment:
FYI: JSON output for `ansible-galaxy collection list` only exists since ansible-core 2.11 (https://github.com/ansible/ansible/commit/67f5bb39c7cf732c4e398318128d599e56ffdf67).

Reply:
Nice catch! Thanks for the heads up!

Reply:
I was wondering "why did I implement manual parsing for this?" (https://github.com/ansible-community/antsibull/blob/main/antsibull/docs_parsing/ansible_doc.py#L184-L201) and then checked - and found out why :)

Reply:
solved!